### ISSUE TITLE
Fix S2259 FP: is not and short-circuit or

### ISSUE THREAD

**Issue by antonioaversa** [SonarSource/sonar-dotnet#7665]
`is not` on a nullable object, followed a short-circuit or, followed by a member access to the nullable object, raises an issue.

```cs
obj?.Value is not Table || (obj.Value is Table); // FP
//                          ^^^

public record Table;
public class ObjectInstance
{
    public object Value { get; }
}
```

Original issue reported [here](https://community.sonarsource.com/t/c-false-positive-object-is-null-on-at-least-one-execution-path/95984).

Reproduced [here](https://github.com/SonarSource/sonar-dotnet/pull/7666).
